packages/cli/src/deploy/ensureWorldFactory.ts

```diff
@@ -6,10 +10,57 @@ import { salt } from "./common";
 import { ensureContractsDeployed } from "./ensureContractsDeployed";
 import { Contract } from "./ensureContract";
 
+export const accessManagementSystemDeployedBytecodeSize = size(
+  accessManagementSystemBuild.deployedBytecode.object as Hex
+);
+export const accessManagementSystemBytecode = encodeDeployData({
+  bytecode: accessManagementSystemBuild.bytecode.object as Hex,
+  abi: [],
+});
+export const accessManagementSystem = getCreate2Address({
+  from: deployer,
+  bytecode: accessManagementSystemBytecode,
+  salt,
+});
+
+export const balanceTransferSystemDeployedBytecodeSize = size(
+  balanceTransferSystemBuild.deployedBytecode.object as Hex
+);
+export const balanceTransferSystemBytecode = encodeDeployData({
+  bytecode: balanceTransferSystemBuild.bytecode.object as Hex,
+  abi: [],
+});
+export const balanceTransferSystem = getCreate2Address({
+  from: deployer,
+  bytecode: balanceTransferSystemBytecode,
+  salt,
+});
+
+export const batchCallSystemDeployedBytecodeSize = size(batchCallSystemBuild.deployedBytecode.object as Hex);
+export const batchCallSystemBytecode = encodeDeployData({
+  bytecode: batchCallSystemBuild.bytecode.object as Hex,
+  abi: [],
+});
+export const batchCallSystem = getCreate2Address({ from: deployer, bytecode: batchCallSystemBytecode, salt });
+
+export const coreRegistrationSystemDeployedBytecodeSize = size(
+  coreRegistrationSystemBuild.deployedBytecode.object as Hex
+);
+export const coreRegistrationSystemBytecode = encodeDeployData({
+  bytecode: coreRegistrationSystemBuild.bytecode.object as Hex,
+  abi: [],
+});
+export const coreRegistrationSystem = getCreate2Address({
+  from: deployer,
+  bytecode: coreRegistrationSystemBytecode,
+  salt,
+});
+
 export const coreModuleDeployedBytecodeSize = size(coreModuleBuild.deployedBytecode.object as Hex);
 export const coreModuleBytecode = encodeDeployData({
   bytecode: coreModuleBuild.bytecode.object as Hex,
-  abi: [],
+  abi: parseAbi(["constructor(address,address,address,address)"]),
```

Comment:
is it possible to get this form an import of the code module? so this can throw a type error when it changes instead of a runtime error if the abi doesn't match

Reply:
Maybe with abitype (we don't have it in cli package yet)
I'd rather not do this as part of this PR, or right now given I wanna move on to codegen refactor

Reply:
I think what he means is importing the typed ABI, i.e.

```ts
import coreModuleAbi from "@latticexyz/world/out/CoreModule.sol/CoreModule.abi.json" assert { type: "json" };

export const coreModuleBytecode = encodeDeployData({
  bytecode: coreModuleBuild.bytecode.object as Hex,
  abi: coreModuleAbi,
  ...
});
```

Reply:
Oh I see, well yeah point is I remember next to no context for this